pages/docs/reference/experimental.md

```diff
@@ -0,0 +1,232 @@
+
+```kotlin
+import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
+// ...
+
+val compileKotlin: KotlinCompile by tasks
+
+compileKotlin.kotlinOptions.freeCompilerArgs += "-Xuse-experimental=org.mylibrary.ExperimentalMarker"
+```
+
+</div>
+</div>
+
+For Maven, it would be:
+
+<div class="sample" markdown="1" theme="idea" data-highlight-only>
+```xml
+<build>
+<plugins>
+    <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-compiler-plugin</artifactId>
+        <version>3.3</version>
+        <configuration>
+            <source>1.8</source>
+            <target>1.8</target>
+            <forceJavacCompilerUse>true</forceJavacCompilerUse>
+            <fork>true</fork>
+            <compilerArgs>
+                <arg>-Xuse-experimental=org.mylibrary.ExperimentalMarker</arg>
+            </compilerArgs>
+        </configuration>
+    </plugin>
+</plugins>
+</build>
+```
+</div>
+
+To accept the usage of multiple experimental APIs on the module level, add one of the described arguments for each experimental API marker used in your module.
+
+## Marking experimental API 
+
+### Creating marker annotations
+
+If you want to declare your module's API as experimental, create an annotation class to use as its _experimental marker_. This class must be annotated with [@Experimental](/api/latest/jvm/stdlib/kotlin/-experimental/index.html):
+
+<div class="sample" markdown="1" theme="idea" data-highlight-only>
+```kotlin
+@Experimental
+@Retention(AnnotationRetention.BINARY)
+@Target(AnnotationTarget.CLASS, AnnotationTarget.FUNCTION)
+annotation class ExperimentalDateTime
+```
+</div>
+
+Experimental marker annotations must meet several requirements:
+* `BINARY` [retention](/api/latest/jvm/stdlib/kotlin.annotation/-annotation-retention/index.html)
+* No `EXPRESSION` and `FILE` among [targets](/api/latest/jvm/stdlib/kotlin.annotation/-annotation-target/index.html)
+* No parameters.
+
+A marker annotation can have one of two severity [levels](/api/latest/jvm/stdlib/kotlin/-experimental/-level/index.html) of informing about experimental API usage:
+* `Experimental.Level.ERROR`. Acceptance is required. Otherwise, the code that uses marked API won't compile. This level is used by default.
+* `Experimental.Level.WARNING`. Acceptance is not required. Without it, the compiler raises a warning.
+To set the desired level, specify the `level` parameter of the `@Experimental` annotation.
+
+<div class="sample" markdown="1" theme="idea" data-highlight-only>
+```kotlin
+@Experimental(level = Experimental.Level.WARNING)
+@Retention(AnnotationRetention.BINARY)
+@Target(AnnotationTarget.CLASS, AnnotationTarget.FUNCTION)
+annotation class ExperimentalDateTime
+```
+</div>
+
+If you publish several features in the experimental state, declare a marker for each. Separate markers make the use of experimental features safer for your clients: they'll be able to use only the features that they explicitly accept.

```

Comment:
Also it makes it easier for you to graduate these features to stable independently.